pageserver/src/waldecoder.rs

```diff
@@ -458,29 +435,29 @@ pub fn decode_wal_record(record: Bytes) -> DecodedWALRecord {
         let block_id = buf.get_u8();
 
         match block_id {
-            XLR_BLOCK_ID_DATA_SHORT => {
+            pg_constants::XLR_BLOCK_ID_DATA_SHORT => {
```

Comment:
Consider `use pg_constants::*;` to avoid having to qualify all the constants.

Reply:
I left it intentionally, just to keep things explicit and differ our zenith constants from postgres ones.

I think we need to isolate code borrowed from Postgres even better because we will have to keep it in sync with upstream changes. Ideally, we should probably just link with postgres headers, but I don't know how to do that.

Reply:
> I left it intentionally, just to keep things explicit and differ our zenith constants from postgres ones.

Ok

> I think we need to isolate code borrowed from Postgres even better because we will have to keep it in sync with upstream changes. Ideally, we should probably just link with postgres headers, but I don't know how to do that.

Yeah, that needs some more thinking. We could expand the use of 'bindgen' to generate Rust constants from the PostgreSQL headers, but the harder problem is to somehow isolate the code that deals with PostgreSQL structs from the rest of the code, so that the Page Server can work with multiple PostgreSQL versions simultaneously.

It's not urgent, though. I think we can ignore that for now, and revisit 12 months from now, when we need to start supporting multiple PostgreSQL major versions. And maybe the solution is that we just compile the page server separately for each supported PostgreSQL version, and run a separate fleet of page servers for each.